code/game/gamemodes/objective.dm

```diff
@@ -17,8 +17,25 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 
 /datum/objective/Destroy()
 	GLOB.all_objectives -= src
+	if(owner)
+		owner.objectives -= src
+		owner = null
 	return ..()
 
+/**
+ * Will try to renew the objective to a valid state with another target.
+ * Returns the new found target or null if none were found
+ */
+/datum/objective/proc/on_target_loss()
+	if(owner?.current)
```

Comment:
Good point. This might however require a rework depending on how much other code relies on the objective not to be null after this call. I'll keep it as is for now but you do raise a good point here.